ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -8174,6 +8176,9 @@ protected Operator genFileSinkPlan(String dest, QB qb, Operator input)
 
     FileSinkOperator fso = (FileSinkOperator) output;
     fso.getConf().setTable(destinationTable);
+    if (destTableIsMaterialization) {
+      ctx.addMaterializedTableSource(destinationTable.getFullTableName(), fso);
```

Comment:
Can this be merged with `Context.addMaterializedTable` and called in `materializeCTE`?
`SemanticAnalyzer` has `getSinkOp()` hence we can get the `FileSinkOperator` later from the analyzer instance.

Does the whole FileSinkOperator is needed?
```
Context.addMaterializedTable(String cteName, Table table, Statistics statistics);
```

Reply:
That sounds cleaner!

> Does the whole FileSinkOperator is needed?

No. I stored the whole operator because stats of FSO are not computed yet at L8180. If we push it to the end of the `materializeCTE`, we can simply store only Statistics.